src/shogun/base/SGObject.h

```diff
@@ -348,39 +348,7 @@ class CSGObject
 	 * @param value value of the parameter
 	 */
 	template <typename T, typename std::enable_if_t<!is_string<T>::value>* = nullptr>
-	void put(const Tag<T>& _tag, const T& value) noexcept(false)
-	{
-		if (has_parameter(_tag))
-		{
-			auto parameter_value = get_parameter(_tag).get_value();
-			if (!parameter_value.cloneable())
-			{
-				SG_ERROR(
-					"Cannot put parameter %s::%s.\n", get_name(),
-					_tag.name().c_str());
-			}
-			try
-			{
-				any_cast<T>(parameter_value);
-			}
-			catch (const TypeMismatchException& exc)
-			{
-				SG_ERROR(
-					"Cannot put parameter %s::%s of type %s, incompatible "
-					"provided type %s.\n",
-					get_name(), _tag.name().c_str(), exc.actual().c_str(),
-					exc.expected().c_str());
-			}
-			ref_value(value);
-			update_parameter(_tag, make_any(value));
-		}
-		else
-		{
-			SG_ERROR(
-				"Parameter %s::%s does not exist.\n", get_name(),
-				_tag.name().c_str());
-		}
-	}
+	void put(const Tag<T>& _tag, const T& value, int64_t step=-1) noexcept(false);
```

Comment:
Since `ObservedValue` needs also a `step` parameter, I've extended `put` such to accept it.

Reply:
no, let's not do that please. `put` is a separate API of the base class, so it shouldnt be influenced by this under-the-hood code